app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -125,8 +125,17 @@ def show
 
     api :POST, "/repositories/:id/sync", N_("Sync a repository")
     param :id, :identifier, :required => true, :desc => N_("repository ID")
+    param :source_url, String, :desc => N_("temporarily override feed URL for sync"), :required => false
     def sync
-      task = async_task(::Actions::Katello::Repository::Sync, @repository)
+      if params[:source_url].present? && params[:source_url] !~ /\A#{URI.regexp}\z/
+        fail HttpErrors::BadRequest, N_("source URL is malformed")
```

Comment:
I think you are correct, it looks like `N_()` is used for the param definitions but not in the actual raised exceptions.